### PR TITLE
OCPCLOUD-2451,OCPBUGS-29832: Bump CAPI 1.6 and k8s 1.28

### PR DESCRIPTION
**Pull request by damdo**
Done:

- bump k8s dependencies 1.28 (aside from the laggard GCP, AWS handled later)
- bump CAPI provider deps to versions compatible to 1.28
- bump CAPI core deps to version compatible to 1.28

### CODE REVIEW

**Comment by JoelSpeed** [comment]
@damdo What is preventing this from moving forward?

File: cmd/cluster-capi-operator/main.go [hunk 1]
```diff
@@ -298,24 +310,9 @@ func setupInfraClusterReconciler(mgr manager.Manager, platform configv1.Platform
...
-	if err := (&webhook.CoreProviderWebhook{}).SetupWebhookWithManager(mgr); err != nil {
-		klog.Error(err, "unable to create webhook", "webhook", "CoreProvider")
-		os.Exit(1)
-	}
-
-	if err := (&webhook.InfrastructureProviderWebhook{Platform: platform}).SetupWebhookWithManager(mgr); err != nil {
-		klog.Error(err, "unable to create webhook", "webhook", "InfrastructureProvider")
-		os.Exit(1)
-	}
-
```
Comment (damdo) [inline cmd/cluster-capi-operator/main.go:311]: These were upstream cluster-api-operator CRs validations. I cleaned those up now that we don't use it anymore.

File: e2e/go.mod [hunk 2]
```diff
@@ -1,79 +1,83 @@
...
-	sigs.k8s.io/cluster-api-provider-gcp v1.5.0
-	sigs.k8s.io/cluster-api-provider-ibmcloud v0.6.0
-	sigs.k8s.io/controller-runtime v0.15.2
-	sigs.k8s.io/yaml v1.3.0
+	k8s.io/api v0.28.6
+	k8s.io/apimachinery v0.28.6
+	k8s.io/client-go v0.28.6
+	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
+	sigs.k8s.io/cluster-api v1.6.2
+	sigs.k8s.io/cluster-api-provider-aws/v2 v2.3.5
```
Comment (damdo) [inline e2e/go.mod:18]: This is still based on the CAPI 1.5.x CAPA release, will update it once 2.4 is out and we did the rebase for the provider fork.

File: e2e/go.mod [hunk 3]
```diff
@@ -1,79 +1,83 @@
...
+	sigs.k8s.io/cluster-api-provider-gcp v1.5.1-0.20240226200925-8c0c11dcff84
```
Comment (damdo) [inline e2e/go.mod:19]: I pinned this to a commit after the CAPI 1.6.x bump, will update it to 1.6.x once a CAPG release is out

**Review by JoelSpeed — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**